README.md

```diff
@@ -97,6 +97,17 @@ Learn more
 ----------
 After you are up and running see the [User Manual](http://fitstool.org/user-manual) for more documentation.
 
+
+Development
+-----------
+
+This project uses a code formatter to apply the [palantir-java-format](https://github.com/palantir/palantir-java-format)
```

Comment:
Thanks for linking to palantir. For example, where can I see whether to use tabs or spaces? and if spaces, how many? Am I allowed to use curly-bracket blocks in one line? Do I need Javadocs on any classes? methods?

Does palantir document these kinds of expectations?

Reply:
Based on the indirect reference to Google styles... maybe this is what I am looking for: https://google.github.io/styleguide/javaguide.html

Reply:
If that is true, linking the Google style-guide in the README would make sense.

Reply:
It's a tweak of Google's formatter which formats to Google's style guide: https://google.github.io/styleguide/javaguide.html

It is not a checkstyle like tool. That is to say it only formats code and does not enforce non-format related style. For example, it doesn't care if you do or don't add javadoc. It only cares that the javadoc that is present is formatted consistently. Likewise, it doesn't care how you name you classes, variables, etc.

The idea is that the developer does not necessarily need to know what the format is. The can write their code with tabs or no spacing at all if they want. They just have to apply the formatter after they're done writing and it will take care of making sure it is consistent with the rest of the code base.

Reply:
Added the link to the google style guide. The reason for using the palantir formatter rather than google's is because palantir's formats lambdas better and allows for slightly longer lines, but it's otherwise the same.

Again, this formatter will never reject code a developer has written and require them to make manual changes, like with checkstyle. If there is ever a check failure, it just means they need to run the formatter and it will be fixed. It is not a tool for enforcing best practices -- just consistent formatting.

Reply:
Thanks. The tools seems great. As you are saying, it does appear that Spotless:check will fail the compilation phase of the build if the formatting is wrong. Hopefully the error message makes it clear to the developer on how to run the command to make the fix.